sql/src/main/java/io/crate/execution/jobs/DistResultRXTask.java

```diff
@@ -156,10 +162,21 @@ private void triggerConsumer() {
             }
             throwable = lastThrowable;
         }
+        final Throwable error = throwable;
         if (invokeConsumer) {
-            consumer.accept(batchPagingIterator, throwable);
+            try {
+                executor.execute(() -> consumer.accept(batchPagingIterator, error));
+            } catch (EsRejectedExecutionException | RejectedExecutionException e) {
```

Comment:
can we replace this by using `ThreadPools.fallbackOnRejection(..)`?

Reply:
Before we aborted the execution, see https://github.com/crate/crate/pull/7467/files#diff-9d5f358fde14becd491b137fd639b169L163

I'd like to keep that behaviour for now. But I've pushed a fixup to delete the RejectedExecution handling that was there.